File: nodejs/base/core/lib/model/service.ts
import { McmaResource, McmaResourceProperties } from "./mcma-resource";
import { ResourceEndpoint, ResourceEndpointProperties } from "./resource-endpoint";

export interface ServiceProperties extends McmaResourceProperties {
    name: string;
    resources: ResourceEndpointProperties[];
    authType?: string;
    authContext?: any;
    jobType?: string;
    jobProfiles?: string[];
}

export class Service extends McmaResource implements ServiceProperties {
    name: string;
    resources: ResourceEndpointProperties[];
    authType?: string;
    authContext?: any;
    jobType?: string;
    jobProfiles?: string[];

    constructor(properties: ServiceProperties) {
        super("Service", properties);

        this.checkProperty("name", "string", true);
        this.checkProperty("resources", "Array", true);
        this.checkProperty("authType", "string", false);
        this.checkProperty("jobType", "string", false);
        this.checkProperty("jobProfiles", "Array", false);

        if (properties.authContext) {
            if (typeof properties.authContext === "string") {
                try {
                    this.authContext = JSON.parse(properties.authContext);
                } catch {
                    this.authContext = properties.authContext;
                }
            } else {
                this.authContext = properties.authContext;
            }
        }

        for (let i = 0; i < this.resources.length; i++) {
            this.resources[i] = new ResourceEndpoint(this.resources[i]);
        }
    }
}

File: dotnet/base/Mcma.Core/Model/Service.cs
using System;
using System.Collections.Generic;

namespace Mcma
{
    public class Service : McmaResource
    {
        public string Name { get; set; }

        public string AuthType { get; set; }

        public string AuthContext { get; set; }

        public ICollection<ResourceEndpoint> Resources { get; set; }

        public string JobType { get; set; }

        public string[] JobProfiles { get; set; }

        public ICollection<Locator> InputLocations { get; set; }

        public ICollection<Locator> OutputLocations { get; set; }
    }
}
